server/jetstream_cluster.go

```diff
@@ -4499,11 +4535,24 @@ func (s *Server) jsClusteredConsumerRequest(ci *ClientInfo, acc *Account, subjec
 		return
 	}
 
-	js.mu.Lock()
-	defer js.mu.Unlock()
+	// Grab our jetstream account info.
+	acc.mu.RLock()
+	jsa := acc.js
+	accName := acc.Name
+	accHaLim := jsa.limits.MaxHaResources
+	acc.mu.RUnlock()
 
 	var resp = JSApiConsumerCreateResponse{ApiResponse: ApiResponse{Type: JSApiConsumerCreateResponseType}}
 
+	if cfg.Durable != _EMPTY_ && js.haExceeded(accName, accHaLim) {
```

Comment:
We should check more when we select the peer group and check that? Not sure cfg.Durable will always be the only pace to check. We can have R>1 ephemerals if attached to a WQ/Interest stream.

Reply:
When in this code R>1. Altered this check to be 
`(cfg.Durable != _EMPTY_ || sa.Config.Retention == WorkQueuePolicy) && js.haExceeded(accName, accHaLim)`